File: apps/backend/src/main/java/com/fittrack/backend/service/AuthService.java
package com.fittrack.backend.service;

import com.fittrack.backend.dto.AuthResponse;
import com.fittrack.backend.dto.LoginRequest;
import com.fittrack.backend.dto.RefreshTokenRequest;
import com.fittrack.backend.dto.RegisterRequest;
import com.fittrack.backend.dto.RequestPasswordResetRequest;
import com.fittrack.backend.dto.ResetPasswordRequest;

public interface AuthService {
    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

    AuthResponse refresh(RefreshTokenRequest request);

    void requestPasswordReset(RequestPasswordResetRequest request);

    void resetPassword(ResetPasswordRequest request);
}

File: apps/frontend/src/services/authService.ts
import { api } from './api'
import type { AuthResponse, LoginRequest, RegisterRequest } from '../types/auth'

export const authService = {
  login: async (payload: LoginRequest) => {
    const { data } = await api.post<AuthResponse>('/api/auth/login', payload)
    return data
  },
  register: async (payload: RegisterRequest) => {
    const { data } = await api.post<AuthResponse>('/api/auth/register', payload)
    return data
  },
  refresh: async (refreshToken: string) => {
    const { data } = await api.post<AuthResponse>('/api/auth/refresh', { refreshToken })
    return data
  },
}

